proxy/IPAllow.cc

```diff
@@ -42,8 +44,64 @@ SignalError(ts::BufferWriter &w, bool &flag)
   }
   Error("%s", w.data());
 }
+
+template <typename... Args>
+void
+ParseError(ts::TextView fmt, Args &&... args)
+{
+  ts::LocalBufferWriter<1024> w;
+  w.printv(fmt, std::forward_as_tuple(args...));
+  w.write('\0');
+  Warning("%s", w.data());
+}
+
 } // namespace
 
+namespace ts
+{
+BufferWriter &
+bwformat(BufferWriter &w, BWFSpec const &spec, IpAllow const *obj)
+{
+  return w.print("{}[{}]", obj->MODULE_NAME, obj->get_config_file());
+}
+
+BufferWriter &
+bwformat(BufferWriter &w, BWFSpec const &spec, YAML::Mark const &mark)
+{
+  return w.print("Line {}", mark.line);
+}
+
+BufferWriter &
```

Comment:
This should probably be moved to `bw_std_format.h` in a separate PR as it's not specific to this PR.